### ISSUE TITLE
Unable to delete member of Project with role Limited Guest

### ISSUE THREAD

**Issue by balonik** [goharbor/harbor#13317]
**Expected behavior and actual behavior:**
I am not able to delete member with role LimitedGuest from any Project.
The Delete from UI or API only produces error:

```
{
  "errors": [
    {
      "code": "NOT_FOUND",
      "message": "project member 270 not found in project 1"
    }
  ]
}
```

This issue appears on instance that was initial deployed as version 1.7.4 and updated to new releases as they become available. I can give exact update path if required.
I was able to identify some discrepancy in Harbor database, see below in Additional context.

**Steps to reproduce the problem:**
1. As user with required permissions, add new member to project with role LimitedGuest.
2. Try to delete the member

**Versions:**
Please specify the versions of following systems.

- harbor version: 2.0.3
- docker engine version: 19.03.12
- docker-compose version: 1.18.0

**Additional context:**
It seems, the role limitedGuest has some strange ID in database. Content of table `role`:

```
registry=# select * from role;
 role_id | role_mask | role_code |     name
---------+-----------+-----------+--------------
       1 |         0 | MDRWS     | projectAdmin
       2 |         0 | RWS       | developer
       3 |         0 | RS        | guest
       4 |         0 | DRWS      | master
      34 |         0 | LRS       | limitedGuest
(5 rows)

registry=# select last_value from role_role_id_seq;
 last_value
------------
         34
(1 row)

```

But all users that are assigned with the limitedGuest role have non-existent role id set. In this case `5`:

```
registry=# select * from project_member where project_id=1;
 id  | project_id | entity_id | entity_type | role |       creation_time        |        update_time
-----+------------+-----------+-------------+------+----------------------------+----------------------------
   1 |          1 |         1 | u           |    1 | 2019-06-13 12:17:45.759919 | 2019-06-13 12:17:45.759919
 270 |          1 |         2 | g           |    5 | 2020-10-20 15:02:28.97263  | 2020-10-20 15:02:28.97263
(3 rows)
```

I have no idea at which point the role limitedGuest get created with `role_id` set to `34`.
I guess the `role_id` set to `5` is something that Harbor expects from fresh install.

As a workaround:
- I could manually update the table `role` in Harbor DB and set `role_id` to `5`. Please confirm this is safe.

As a fix:
- Harbor will get a patch where it will look for actual `role_id` instead of hardcoded values.

**Comment by balonik** [comment]
As one-time fix I have updated the `project_member` table:

```
update project_member set role=34 where project_id=1 and id=270;
```

After that I was able to delete the member of project.

**Comment by xaleeks** [comment]
you're not supposed to have to have to do DB update to delete a user

@heww  can you check into this?

**Comment by heww** [comment]
Hi, @balonik , update the role_id of limitedGuest to 5 is the right way to fix this issue. I will create a PR to fix this in v2.2.
